File: frontend/lib/models/payment.dart
class Payment {
  final int id;
  final int workerId;
  final String? workerName;
  final int? taskId;
  final String? taskTitle;
  final double amount;
  final String status;
  final String date;
  final String? paidAt;

  Payment({
    required this.id,
    required this.workerId,
    this.workerName,
    this.taskId,
    this.taskTitle,
    required this.amount,
    required this.status,
    required this.date,
    this.paidAt,
  });

  factory Payment.fromJson(Map<String, dynamic> json) {
    return Payment(
      id: json['id'],
      workerId: json['worker_id'],
      workerName: json['worker_name'],
      taskId: json['task_id'],
      taskTitle: json['task_title'],
      amount: (json['amount'] as num).toDouble(),
      status: json['status'],
      date: json['date'],
      paidAt: json['paid_at'],
    );
  }

  bool get isPaid => status == 'paid';
  bool get isUnpaid => status == 'unpaid';
}

File: backend/models/payment.py
from utils.db import db
from datetime import datetime

class Payment(db.Model):
    __tablename__ = 'payments'
    
    id = db.Column(db.Integer, primary_key=True)
    worker_id = db.Column(db.Integer, db.ForeignKey('users.id'), nullable=False)
    task_id = db.Column(db.Integer, db.ForeignKey('tasks.id'), nullable=True)
    amount = db.Column(db.Float, nullable=False)
    status = db.Column(db.String(20), nullable=False, default='unpaid')  # unpaid, paid
    date = db.Column(db.DateTime, default=datetime.utcnow)
    paid_at = db.Column(db.DateTime, nullable=True)
    
    def to_dict(self):
        """Convert payment to dictionary"""
        return {
            'id': self.id,
            'worker_id': self.worker_id,
            'worker_name': self.worker.full_name if self.worker else None,
            'task_id': self.task_id,
            'task_title': self.task.title if self.task else None,
            'amount': self.amount,
            'status': self.status,
            'date': self.date.isoformat(),
            'paid_at': self.paid_at.isoformat() if self.paid_at else None
        }
    
    def __repr__(self):
        return f'<Payment {self.id} - {self.status}>'
